app/src/main/java/com/folen/androidshowreel/activity/FragmentBackstackActivity.java

```diff
@@ -0,0 +1,113 @@
+        mToolbar.setTitle("Fragments Backstack");
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.toolbar_menu_backstack, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.action_add:
+                fragment = new FirstFragment();
+                addFragment(fragment);
+                break;
+            case R.id.action_replace:
+                fragment = new SecondFragment();
+                replaceFragment(fragment);
+                break;
+            case R.id.action_add_backstack:
+                fragment = new FirstFragment();
+                addFragmentWithBackStack(fragment);
+                break;
+            case R.id.action_replace_backstack:
+                fragment = new SecondFragment();
+                replaceFragmentWithBackStack(fragment);
+                break;
+            case R.id.action_remove:
+                removeFragment(fragment);
+                break;
+            case android.R.id.home:
+                removeFragment(fragment);
+                break;
+        }
+        return true;
+    }
+
+    public void addFragment(Fragment fragment) {
+        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
+        fragmentTransaction.add(R.id.fragment_content, fragment);
+        fragmentTransaction.commit();
+    }
+
+    public void replaceFragment(Fragment fragment) {
+        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
+        fragmentTransaction.replace(R.id.fragment_content, fragment);
+        fragmentTransaction.commit();
+    }
+
+    public void addFragmentWithBackStack(Fragment fragment) {
+        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
+        fragmentTransaction.add(R.id.fragment_content, fragment);
+        fragmentTransaction.addToBackStack(null);
+        fragmentTransaction.commit();
+    }
+
+    public void replaceFragmentWithBackStack(Fragment fragment) {
+        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
+        fragmentTransaction.replace(R.id.fragment_content, fragment);
+        fragmentTransaction.addToBackStack(null);
+        fragmentTransaction.commit();
+    }
+
+    public void removeFragment(Fragment fragment) {
+        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
+        fragmentTransaction.remove(fragment);
+        fragmentTransaction.commit();
+    }
+
+    public static Intent intent(Context context) {
+        return new Intent(context, FragmentBackstackActivity.class);
+    }
+
+    @Override
+    public void onBackPressed(){
+        getSupportFragmentManager().popBackStack();
```

Comment:
ta komenda wypierdala cala historie backstacka a o to chyba nam nie chodzilo? tutaj raczej nie bedzie potrzeby overidowania  onBackPressed w tym zadaniu